### ISSUE TITLE
Windows.KapeFiles.Extract is not working in 0.6.7

### ISSUE THREAD

**Issue by nullifysecurity** [Velocidex/velociraptor#2273]
**Version:** 0.6.7

**What's the issue?**

Due to the offline collector changes in version 0.6.7 the `Windows.KapeFiles.Extract` artifact can no longer find the correct paths for the `Uploads.json` and `All File Metadata.json` when recreating the timestamps in the KAPE extract archive (.zip).

**How to replicate?**

1. Create a new offline collector for `Windows.KapeFiles.Targets` in the Velociraptor console
2. Download and execute the collector on the target host
3. Run `Windows.KapeFiles.Extract` from the command-line specifying the collected KAPE extract archive to recreate the timestamps

**Potential solution?**

Modify the `Windows.KapeFiles.Targets` artifact to use the `collector` accessor and point to the new `Uploads.json` path generated by the collector to recreate the timestamps from the KAPE collection.

**Comment by nullifysecurity** [comment]
It appears that the `uploads.json` file holds the original path in `vfs_path` and the actual path in the archive in `_Components`. The `vfs_path` looks like it matches up nicely with the `SourceFile` in `results/Windows.KapeFiles.Targets%2FAll File Metadata.json`.

Would need to look at the path in `_Components` to get the files, then do a lookup for where `vfs_path` equals `SourceFile` in the metadata to then read the actual timestamps and use these in the `upload_directory` function.

**Comment by scudette** [comment]
Thanks for reporting this issue. I just submitted a fix and a test for it. Can you please check out the latest ci build and see if it's working for you?

**Comment by nullifysecurity** [comment]
> Thanks for reporting this issue. I just submitted a fix and a test for it. Can you please check out the latest ci build and see if it's working for you?

Good morning @scudette, 

Confirming that the fix in the latest CI build (https://github.com/Velocidex/velociraptor/actions/runs/3548402006) fixes the `Windows.KapeFiles.Extract` issue with KAPEs collected with an offline collector. However, it does not work with a KAPE collected from the Velociraptor GUI using `Windows.KapeFiles.Targets`. Is it possible to modify the artifact to have it work with both?

When using the `Windows.KapeFiles.Extract` artifact on a KAPE taken via the Velociraptor GUI, the files have the following "Zip: Not found:" error.

```
{
    "Timestamp": 1669595167,
    "started": "2022-11-28 00:26:07.7640272 +0000 UTC",
    "vfs_path": "C:\\Windows\\System32\\config\\SYSTEM.LOG2.idx",
    "_Components": [
        "clients",
        "C.72baadf00eff8ec4",
        "collections",
        "F.CE1VVM7VB65QG",
        "uploads",
        "auto",
        "C:",
        "Windows",
        "System32",
        "config",
        "SYSTEM.LOG2"
    ],
    "file_size": 260,
    "uploaded_size": 260,
    "FileUpload": "/clients/C.72baadf00eff8ec4/collections/F.CE1VVM7VB65QG/uploads/auto/C%3A/Windows/System32/config/SYSTEM.LOG2",
    "Dest": "/F.CE1VVM7VB65QG/uploads/auto/C:/Windows/System32/config/SYSTEM.LOG2",
    "Metadata": null,
    "UploadedFile": {
        "Path": "",
        "Size": 0,
        "Error": "Zip: Not found: /clients/C.72baadf00eff8ec4/collections/F.CE1VVM7VB65QG/uploads/auto/C%3A/Windows/System32/config/SYSTEM.LOG2."
    }
}
```

**Development Build Details**
```
{
    "name": "velociraptor",
    "version": "0.6.8-dev",
    "commit": "49c8ad8",
    "build_time": "2022-11-25T13:42:15Z",
    "ci_build_url": "https://github.com/Velocidex/velociraptor/actions/runs/3548402006",
    "compiler": "go1.19.3"
}
```

**Comment by scudette** [comment]
Just submitted #2282 which replicates the offline collector's upload.json from the flow download action. Can you please give it a test? You can re-export the collection from the gui and see if the extract artifact works better with it :-)

**Comment by nullifysecurity** [comment]
Good afternoon @scudette,

Confirming this issue is now resolved on the build with commit 0da923d, both GUI and offline collector KAPE collections are now being successfully parsed by the `Windows.KapeFiles.Extract` artifact.

Thank you for your efforts!

**Development Build Details**

```json
{
    "name": "velociraptor",
    "version": "0.6.8-dev",
    "commit": "0da923d",
    "build_time": "2022-11-29T07:32:58Z",
    "ci_build_url": "https://github.com/Velocidex/velociraptor/actions/runs/3571927452",
    "compiler": "go1.19.3"
}
```
